packages/typespec-client-generator-core/src/package.ts

```diff
@@ -0,0 +1,734 @@
+import { getLroMetadata, getPagedResult } from "@azure-tools/typespec-azure-core";
+import {
+  Operation,
+  Type,
+  UsageFlags,
+  getNamespaceFullName,
+  getService,
+  ignoreDiagnostics,
+  isErrorModel,
+} from "@typespec/compiler";
+import {
+  HttpOperation,
+  getHeaderFieldName,
+  getHttpOperation,
+  getServers,
+  isContentTypeHeader,
+} from "@typespec/http";
+import { resolveVersions } from "@typespec/versioning";
+import {
+  getAccess,
+  listClients,
+  listOperationGroups,
+  listOperationsInOperationGroup,
+} from "./decorators.js";
+import {
+  SdkBodyParameter,
+  SdkClient,
+  SdkClientType,
+  SdkContext,
+  SdkEndpointParameter,
+  SdkEnumType,
+  SdkHeaderParameter,
+  SdkHttpOperation,
+  SdkHttpResponse,
+  SdkInitializationType,
+  SdkLroPagingServiceMethod,
+  SdkLroServiceMethod,
+  SdkMethod,
+  SdkMethodParameter,
+  SdkMethodResponse,
+  SdkModelPropertyType,
+  SdkModelType,
+  SdkOperationGroup,
+  SdkPackage,
+  SdkPagingServiceMethod,
+  SdkParameter,
+  SdkPathParameter,
+  SdkQueryParameter,
+  SdkServiceMethod,
+  SdkServiceOperation,
+  SdkServiceParameter,
+  SdkServiceResponseHeader,
+  SdkType,
+} from "./interfaces.js";
+import {
+  getAvailableApiVersions,
+  getClientNamespaceStringHelper,
+  getDocHelper,
+  getHashForType,
+  getSdkTypeBaseHelper,
+  isAcceptHeader,
+  updateWithApiVersionInformation,
+} from "./internal-utils.js";
+import {
+  getClientNamespaceString,
+  getDefaultApiVersion,
+  getLibraryName,
+  getPropertyNames,
+} from "./public-utils.js";
+import {
+  addEncodeInfo,
+  addFormatInfo,
+  getAllModels,
+  getClientType,
+  getSdkCredentialParameter,
+  getSdkModelPropertyType,
+} from "./types.js";
+
+function getSdkHttpBodyParameters(
+  context: SdkContext<SdkHttpOperation>,
+  httpOperation: HttpOperation,
+  methodParameters: SdkParameter[]
+): SdkBodyParameter[] | undefined {
+  const tspBody = httpOperation.parameters.body;
+  if (tspBody === undefined) return undefined;
+  let contentTypes = tspBody.contentTypes;
+  if (contentTypes.length === 0) {
+    contentTypes = ["application/json"];
+  }
+  const defaultContentType = contentTypes.includes("application/json")
+    ? "application/json"
+    : contentTypes[0];
+  if (!tspBody.parameter) {
+    const bodyType = getClientType(context, tspBody.type, httpOperation.operation);
+    return [
+      {
+        kind: "body",
+        nameInClient: "body",
+        description: getDocHelper(context, tspBody.type).description,
+        details: getDocHelper(context, tspBody.type).details,
+        onClient: false,
+        contentTypes: contentTypes,
+        defaultContentType: defaultContentType,
+        isApiVersionParam: false,
+        apiVersions: getAvailableApiVersions<SdkHttpOperation>(context, tspBody.type),
+        type: bodyType,
+        optional: false,
```

Comment:
I suspect this is the culprit that's making body params always required?

Reply:
yeah, i'm trying to figure out how to find out if a body param is optional though, bc there's not that property on the tsp body param

Reply:
I added a test for this and I see that the method param is correctly marked as optional, it's just the operation param has it as required.